src/services/LoggingService.js

```diff
@@ -3,36 +3,67 @@
+    const processedError = Object.create(error);
+    processedError.message = processedError.message.substring(0, MAX_ERROR_LENGTH);
+    return processedError;
+  } else if (typeof error === 'string' && error.length > MAX_ERROR_LENGTH) {
+    return error.substring(0, MAX_ERROR_LENGTH);
+  }
+  return error;
+}
+
 class LoggingService {
   static logInfo(message) {
     if (typeof newrelic !== 'undefined') {
       newrelic.addPageAction('INFO', { message });
     }
   }
 
-  static logError(error) {
+  static logError(error, customAttributes) {
     if (typeof newrelic !== 'undefined') {
       // Note: customProperties are not sent.  Presumably High-Security Mode is being used.
-      newrelic.noticeError(error);
+      newrelic.noticeError(fixErrorLength(error), customAttributes);
     }
   }
 
-  // Note: will simply log errors that don't seem to be API error responses.
-  static logAPIErrorResponse(error) {
-    let processedError;
+  // API errors look for axios API error format.
+  // Note: function will simply log errors that don't seem to be API error responses.
+  static logAPIErrorResponse(error, customAttributes) {
+    let processedError = error;
+    let updatedCustomAttributes = customAttributes;
+
     if (error.response) {
-      processedError = new Error(`API request failed: ${error.response.status} ${error.response.config.url} ${JSON.stringify(error.response.data)}`);
+      let data = !error.response.data ? '' : JSON.stringify(error.response.data);
+      // Don't include data if it is just an HTML document, like a 500 error page.
+      data = data.includes('<!DOCTYPE html>') ? '' : data;
+      const responseAttributes = {
+        errorType: 'api-response-error',
+        errorStatus: error.response.status,
+        errorUrl: error.response.config ? error.response.config.url : '',
+        errorData: data,
+      };
+      updatedCustomAttributes = Object.assign({}, customAttributes, responseAttributes);
+      processedError = new Error(`API request failed: ${error.response.status} ${responseAttributes.errorUrl} ${data}`);
     } else if (error.request) {
+      const requestAttributes = {
+        errorType: 'api-request-error',
+        errorStatus: error.request.status,
+        errorUrl: error.request.responseURL,
+        errorData: error.request.responseText,
+      };
+      updatedCustomAttributes = Object.assign({}, customAttributes, requestAttributes);
       processedError = new Error(`API request failed: ${error.request.status} ${error.request.responseURL} ${error.request.responseText}`);
-    } else {
-      processedError = Object.create(error);
     }
-    if (processedError.message) {
-      // NewRelic will not log the error if it is too long.
-      processedError.message = processedError.message.substring(0, 4000);
-    }
-    this.logError(processedError);
+
+    this.logError(processedError, updatedCustomAttributes);
   }
 }
 
 export default LoggingService;
+export { MAX_ERROR_LENGTH };
```

Comment:
Reviewers: Is it wrong to export only for a test?  Should I just duplicate the constant?

Reply:
If we're being super-purists about it then we'd say it's wrong. :)  Practically speaking, no one will ever really notice.  If if were in a separate constants file and imported here, then it wouldn't feel as wrong, but I wouldn't bother.